clients/src/main/java/org/apache/kafka/clients/telemetry/DefaultClientTelemetry.java

```diff
@@ -398,6 +404,46 @@ else if (state == TelemetryState.terminating_push_in_progress)
             throw new IllegalTelemetryStateException(String.format("Could not transition state after successful push telemetry from state %s", state));
     }
 
+    /**
+     * Examine the response data and handle different error code accordingly.
+     *  - Authorization Failed: Retry 30min later
+     *  - Invalid Record: Retry 5min later
+     *  - UnknownSubscription or Unsupported Compression: Retry
+     *  After a new subscription is created, the current state is transitioned to push_needed to wait for another push.
+     *
+     * @param data pushTelemetryResponseData
+     * @throws IllegalTelemetryStateException when subscription is mull
+     */
+    private void handlePushTelemetryResponseDataErrors(final PushTelemetryResponseData data) {
+        TelemetrySubscription currentSubscription = subscription().orElseThrow(
+        () -> new IllegalTelemetryStateException(String.format("Could not transition state after successful push telemetry from state %s", state)));
```

Comment:
This error seems a little misleading since this push telemetry *wasn't* successful. Plus we're not transitioning state at that point, just getting the subscription. The error is that we must have a subscription if we're responding to a push event because the subscription is retrieved before the push ever happens.

Reply:
I think I copied your throw from a different place.  In fact, it should be noting the caller about null subscription.